### PR TITLE
Adds Machine Config Controller cli

### PR DESCRIPTION
**Pull request by abhinavdahiya**
* adds `MachineConfigController` that runs following sub controllers
  * template sub controller
  * render sub controller
  * node update sub controller
* examples for CRDs

based on #5

### CODE REVIEW

**Comment by abhinavdahiya** [comment]
@yifan-gu rebased after #5 merged. PTAL

**Comment by yifan-gu** [comment]
The commit title of the 2nd to last commit has a typo: `s/remane/rename`

File: cmd/machine-config-controller/main.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+		Long:  "",
+	}
+)
+
+func init() {
+	rootCmd.PersistentFlags().AddGoFlagSet(flag.CommandLine)
+}
+
+func main() {
+
```
Comment (yifan-gu) [inline cmd/machine-config-controller/main.go]: nit: kill the empty line?

Comment (abhinavdahiya) [reply cmd/machine-config-controller/main.go]: done

**Pull request merged**